src/runtime/DateObject.cpp

```diff
@@ -1419,4 +1419,50 @@ void* DateObject::operator new(size_t size)
     }
     return GC_MALLOC_EXPLICITLY_TYPED(size, descr);
 }
+
+Value DateObject::makeDay(ExecutionState& state, const Value& year, const Value& month, const Value& day)
+{
+    if (!std::isfinite(year.asNumber()) && !std::isfinite(month.asNumber()) && !std::isfinite(day.asNumber())) {
+        return Value(std::numeric_limits<double>::quiet_NaN());
+    }
+
+    int y = year.asInt32();
```

Comment:
You tested type before as Number
but read asInt32
Int32 and Number type is different
You can test type as isInt32 or you can use toInt32()